x-pack/plugin/ent-search/src/test/java/org/elasticsearch/xpack/application/connector/ConnectorIndexServiceTests.java

```diff
@@ -37,8 +41,73 @@ public void testPutConnector() throws Exception {
 
         DocWriteResponse resp = awaitPutConnector(connector);
         assertThat(resp.status(), anyOf(equalTo(RestStatus.CREATED), equalTo(RestStatus.OK)));
+    }
+
+    public void testDeleteConnector() throws Exception {
+        int numConnectors = 5;
+        List<String> connectorIds = new ArrayList<>();
+        for (int i = 0; i < numConnectors; i++) {
+            Connector connector = ConnectorTestUtils.getRandomConnector();
+            connectorIds.add(connector.getConnectorId());
+            DocWriteResponse resp = awaitPutConnector(connector);
+            assertThat(resp.status(), equalTo(RestStatus.CREATED));
+        }
+
+        String connectorIdToDelete = connectorIds.get(0);
+        DeleteResponse resp = awaitDeleteConnector(connectorIdToDelete);
+        assertThat(resp.status(), equalTo(RestStatus.OK));
+        expectThrows(ResourceNotFoundException.class, () -> awaitGetConnector(connectorIdToDelete));
 
-        // TODO: more checks once GET endpoint is implemented :)
+        expectThrows(ResourceNotFoundException.class, () -> awaitDeleteConnector(connectorIdToDelete));
+    }
+
+    // TODO: figure out how to use ConnectorTemplateRegistry in the test node
```

Comment:
Given that `listConnectors` endpoint is already comprehensively tested in YAML tests I would prefer to remove this test for this PR. 

The issue is that I'm so far unsuccessful at registering connector index templates (with default mappings) therefore this test fails for the list operation, since it expects the `connector_id` to be an indexed field.

I will ask/look around to get the templates registered for such a scenario, I couldn't find examples in the codebase.